import { Model54, make54 } from "./module-0054";

export interface Model55<T> {
  readonly current: T;
  readonly previous: Model54<T>;
}

export function make55<T>(value: T): Model55<T> {
  return { current: value, previous: make54(value) };
}

export const marker55: Model55<number> = make55(55);
