class Service335 {
  id = 335;
  label = "service-335";
  meta = { active: false, score: 335 };
}
class Alternate335 {
  id = 335;
  label = 335;
  meta = { active: false, score: 335 };
}
module.exports = new Service335();
module.exports = new Alternate335();
