(function () {"use strict";
// zig-out/ipc-handles-port-2026-10-09/generated-pinned-net/InternalAsyncHooksTick.mts
var $, tickInitHooks = [], nextAsyncId = 1;
$ = {
  tickInitHooks,
  newAsyncId() {
    return ++nextAsyncId;
  }
};
return $

})
